### PR TITLE
Patch CVE-2019-17498

### PR DESCRIPTION
**Pull request by TheEvilSkeleton**
CVE: https://nvd.nist.gov/vuln/detail/CVE-2019-17498

The commit I used to patch it: https://github.com/libssh2/libssh2/commit/dedcbd106f8e52d5586b0205bc7677e4c9868f9c

We'll need to test if everything works.

### CODE REVIEW

**Comment by TheEvilSkeleton** [comment]
It works fine for me. I'm not sure if it'll work for others or not.

**Comment by albfan** [comment]
Is this fixed in any libssh2 release? I prefer to move to an existing release.

For that open this bug on gitg repo is a more easy way to handle this

**Comment by TheEvilSkeleton** [comment]
> Is this fixed in any libssh2 release? I prefer to move to an existing release.

No. The version we're using in libssh2 is the latest version. The patch I used came from a commit that happened after the last stable release.

**Comment by albfan** [comment]
No, I'm not libssh2 maintainer.

**Comment by TheEvilSkeleton** [comment]
Not to sound rude, but what does being a libssh2 maintainer have to do with all of that? I'm only here to apply a patch on a year old vulnerability, it has nothing to do with being a libssh2 maintainer.

**Comment by albfan** [comment]
I trust in libssh2 to provide safe and reliable releases, patching them will only turn me into a libssh2 maintainer here. Have no idea what the implications of that patch are, and honestly as long as ssh works I don't care.

If you show me other downstream packagers bundled that patch because is critical (but I can't see how it can be critical if libssh2 do not release a 1.9.1 or 1.10.0 version) probably I would consider to add it, as flatpak is indeed like a like distro.

For what I care (arch linux is my distro) https://github.com/archlinux/svntogit-packages/blob/packages/libssh2/trunk/PKGBUILD they don't patch libssh2, why should I do it here?

Probably you need to open an issue in libssh so they release a bugfix version (1.9.1, 1.10.0) and then suggest a bump of libssh2 here.
 
Is that fix important? probably. Is here the place to apply it? no.

**Comment by TheEvilSkeleton** [comment]
> Is that fix important? probably. Is here the place to apply it? no.

Fair point.

**Comment by TheEvilSkeleton** [comment]
> If you show me other downstream packagers bundled that patch because is critical (but I can't see how it can be critical if libssh2 do not release a 1.9.1 or 1.10.0 version) probably I would consider to add it, as flatpak is indeed like a like distro.

Actually, openSUSE, Debian, Gentoo and Fedora use a patched libssh2:

- http://lists.opensuse.org/opensuse-security-announce/2019-11/msg00026.html
- https://lists.debian.org/debian-lts-announce/2019/11/msg00010.html
- https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/TY7EEE34RFKCTXTMBQQWWSLXZWSCXNDB/
- https://gitweb.gentoo.org/repo/gentoo.git/tree/net-libs/libssh2/libssh2-1.9.0_p20200614.ebuild#n8

**Comment by albfan** [comment]
Please fix upstream https://github.com/libssh2/libssh2/issues/539 then bump release.

Any suggestion about Security CVE @barthalion?

**Comment by TheEvilSkeleton** [comment]
@albfan yeah I know. I just wanted to inform you since you asked downstream packagers that patch it.

**Comment by albfan** [comment]
Sure. Is just I'm not sure about flatpak policies about security.

Going down that road is an endless workload.

Not really sure what happen with that libssh2 1.9.1 release.

**Comment by TheEvilSkeleton** [comment]
I pinged the maintainer and asked if there are any updates.

**Comment by barthalion** [comment]
It's up to application maintainer what is considered worth patching. Personally I would do it, given the patch comes from upstream and should apply cleanly.

**Comment by albfan** [comment]
Makes sense @barthalion. For whatever reason libssh2 is stuck.

@TheEvilSkeleton will review in a while

**Comment by TheEvilSkeleton** [comment]
@albfan review as in PR with a patched commit? Not sure what review means in this context, sorry.

File: org.gnome.gitg.json [hunk 1]
```diff
@@ -161,7 +165,7 @@
             },
             {
                 "type": "patch",
-                "path": "appdata.patch"
+                "path": "patches/0001-appdata.patch"
```
Comment (albfan) [inline org.gnome.gitg.json:168]: Don't move to patches for.

Root dir is ok.

File: org.gnome.gitg.json [hunk 2]
```diff
@@ -50,6 +50,10 @@
                 "type": "archive",
                 "url": "https://www.libssh2.org/download/libssh2-1.9.0.tar.gz",
                 "sha256": "d5fb8bd563305fd1074dda90bd053fb2d29fc4bce048d182f96eaa466dfadafd"
+            },
+            {
+                "type": "patch",
+                "path": "patches/0002-CVE-2019-17498.patch"
```
Comment (albfan) [inline org.gnome.gitg.json:56]: Use root dir

File: patches/0002-CVE-2019-17498.patch [hunk 3]
```diff
@@ -0,0 +1,257 @@
+diff --git a/src/packet.c b/src/packet.c
```
Comment (albfan) [inline patches/0002-CVE-2019-17498.patch:1]: Use format patch to extract commit message from libssh2.

We don't want anything but pristine copy of that change.

**Review by albfan — COMMENTED** [review]
This needs to be a single commit

**Comment by TheEvilSkeleton** [comment]
@albfan I'll submit a new MR because I deleted my branch a while ago.

**Comment by albfan** [comment]
Finally there's a new release for libssh2. We will update the dependency in a new release

https://www.libssh2.org/mail/libssh2-devel-archive-2021-08/0001.shtml

**Pull request closed without merging**